propulate/propagators/pso/canonical.py

```diff
@@ -0,0 +1,34 @@
+from typing import List
+
+import numpy as np
+
+from propulate.particle import Particle
+from constriction import Constriction
+
+
+class Canonical(Constriction):
+    """
+    This propagator subclass features a combination of Constriction PSO and VelocityClamping PSO.
+
+    The velocity clamping is made with a clamping factor of 1,
+    the constriction is done as on the parental Constriction propagator.
+
+    For information on the method parameters, please refer to the Constriction propagator.
+    """
+
+    def __init__(self, c_cognitive, c_social, rank, limits, rng):
+        super().__init__(c_cognitive, c_social, rank, limits, rng)
+        x_min, x_max = self.laa
+        x_range = np.abs(x_max - x_min)
+        self.v_cap: np.ndarray = np.array([-x_range, x_range])
```

Comment:
I think this is not what is described in the original paper. However, the stuff described with $V_{max}$ and $X_{max}$ in the original paper is not very reasonable and useful as it seems to hold for symmetric search spaces centered around 0 only. Maybe we should consider putting a factor of 0.5 to scale ``xrange`` here:

R. C. Eberhart and Y. Shi, "Comparing inertia weights and constriction factors in particle swarm optimization," Proceedings of the 2000 Congress on Evolutionary Computation. CEC00 (Cat. No.00TH8512), La Jolla, CA, USA, 2000, pp. 84-88 vol.1, doi: 10.1109/CEC.2000.870279.

Reply:
Nah, when $v_{max}$ is set to $x_{max}$, then the formula used in the code is correct.